bsip-0044.md

```diff
@@ -0,0 +1,173 @@
+
+### **Escrow Asset**
+
+An HTLC involves a conditional transfer of the defined `asset symbol` in the amount of `assets quantity` from the `depositor` to the `recipient`. The HTLC holds these designated `escrow assets` from `depositor` on the blockchain and will continue to enforce the specified `conditions` until one is satisfied.
+
+### **Conditions**
+
+There are two competing conditions within an HTLC, the `hash lock` and the `time lock`.
+
+The HTLC contains a `hash lock` condition, which comprise both the `preimage hash` and `length of the preimage`, barring the transfer of held `escrow assets` unless satisfied. If a `preimage` of requisite `length` is provided to the HTLC which generates a hash matching the `preimage hash`, the `preimage` is then stored within the blockchain, and the `escrow assets` are transferred to the `recipient`.
+
+If a satisfactory `preimage` is not provided to the HTLC before the stipulated `time lock` expires, the `depositor` may request the return of `escrow assets`. The HTLC will only evaluate transfer request from `depositor` and after `timeout threshold`, then return `escrow assets` to `depositor`.
+
+### **Condition Evaluators**
+
+The `preimage` can be thought of a secret key, that will eventually be shared with the `recipient`. This can be a word, a phrase, or even a random series of bytes. The `length` of the `preimage` must be specified within the HTLC at creation.
+
+Upon presentation of a `preimage`, the HTLC `condition evaluator` validates:
+
+1. That the `timeout threshold` has not yet occurred.
+
+2. That the length of the `preimage` matches the specified `preimage length`. 
+
+3. That the `hash of the preimage` matches the specified `preimage hash`. 
+
+If all evaluations succeed, the `escrow assets` are transferred to the `recipient`. If any evaluation fails, nothing happens; the HTLC remains ready to evaluate the next `preimage`.
+
+Upon presentation of a request from the `depositor` for return of `escrow assets`, the HTLC `condition evaluator` validates that the `timeout threshold` is now in the past. It then returns the `escrow assts` to `depositor`.
+
+### **Timing of Condition Evaluation**
+
+The `timeout threshold` of the contract is defined by `depositor` within the HTLC at creation. It can be any time in the future and should allow enough time for `recipient` to review the HTLC and provide the `preimage`. Further, it should not be set too far into the future to mitigate against an unresponsive `recipient` impacting `depositor`, as their `escrow assets` will be locked until `timeout threshold` expiry. The accuracy is based on when the `condition evaluator` runs, and should be considered accurate ± 15 seconds.
```

Comment:
I'd suggest that a committee-configurable maximum period is implemented. The lockup potentially consumes memory resources until the timeout, therefore we should put a limit upon it.

Reply:
I will agree with the value in allowing the Committee to set and maintain the maximum timeout threshold. Consideration should be made regarding the timeout to ensure the needs of those offering "layer 2" services can maintain "payment channels" of reasonable length. Perhaps the timeout on an existing contract could be "extended" by paying an additional fee in a update operation, rather than having to establish a new HTLC.

Reply:
I had thought about a time extension during the initial stages of planning. I decided it was one more thing that would probably rarely be used and we would have to code for. I believe it is something we can add later. IMO: It would not be hard to do, I just do not know its value.